edalize/spyglass.py

```diff
@@ -0,0 +1,113 @@
+import logging
+import os.path
+import platform
+import re
+
+from edalize.edatool import Edatool
+
+logger = logging.getLogger(__name__)
+
+""" Synopsys (formerly Atrenta) Spyglass Backend
+
+Spyglass performs static source code analysis on HDL code and checks for common
+coding errors or coding style violations.
+
+Example snippet of a CAPI2 description file:
+
+spyglass:
+  methodology: "GuideWare/latest/block/rtl_handoff"
+  goals:
+    - lint/lint_rtl
+  spyglass_options:
+    # prevent error SYNTH_5273 on generic RAM descriptions
+    - handlememory yes
+
+"""
+class Spyglass(Edatool):
+
+    tool_options = {
+        'members' : {
+            'methodology' : 'String'
+        },
+        'lists': {
+            'goals': 'String',
+            'spyglass_options': 'String',
+        },
+    }
+
+    argtypes = ['vlogdefine', 'vlogparam']
+
+    """ Configuration is the first phase of the build
+
+    This writes the project TCL files and Makefile. It first collects all
+    sources, IPs and contraints and then writes them to the TCL file along
+     with the build steps.
+    """
+    def configure_main(self):
+        (src_files, incdirs) = self._get_fileset_files(force_slash=True)
+
+        self.jinja_env.filters['src_file_filter'] = self.src_file_filter
+
+        has_systemVerilog = False
+        for src_file in src_files:
+            if src_file.file_type.startswith('systemVerilogSource'):
+                has_systemVerilog = True
+                break
+
+        template_vars = {
+            'name'              : self.name,
+            'src_files'         : src_files,
+            'incdirs'           : incdirs,
+            'tool_options'      : self.tool_options,
+            'toplevel'          : self.toplevel,
+            'vlogparam'         : self.vlogparam,
+            'vlogdefine'        : self.vlogdefine,
+            'has_systemVerilog' : has_systemVerilog,
+            'sanitized_goals'   : [],
+        }
+
+        self.render_template('spyglass-project.prj.j2',
+                             self.name + '.prj',
+                             template_vars)
+
+        # Create a single TCL file for each goal
+        goals = ['Design_Read'] + self.tool_options['goals']
```

Comment:
FuseSoC crashes if goals is left out. Add a check and print raise a RuntimeError or set a sensible default if it's required